src/util/RtreeBasicGeometry.h

```diff
@@ -0,0 +1,171 @@
+  // Create a bounding box, based on the corner coordinates
+  static BasicGeometry::BoundingBox CreateBoundingBox(double pointOneX,
+                                                      double pointOneY,
+                                                      double pointTwoX,
+                                                      double pointTwoY) {
+    return {{pointOneX, pointOneY}, {pointTwoX, pointTwoY}};
+  }
+
+  // ___________________________________________________________________________
+  // Take two bounding boxes and combine them into one bounding box containing
+  // both
+  static BasicGeometry::BoundingBox CombineBoundingBoxes(
+      const BasicGeometry::BoundingBox& b1,
+      const BasicGeometry::BoundingBox& b2) {
+    auto minX = [](BasicGeometry::BoundingBox b) -> double {
+      return b.min_corner().get<0>();
+    };
+    auto minY = [](BasicGeometry::BoundingBox b) -> double {
+      return b.min_corner().get<1>();
+    };
+    auto maxX = [](BasicGeometry::BoundingBox b) -> double {
+      return b.max_corner().get<0>();
+    };
+    auto maxY = [](BasicGeometry::BoundingBox b) -> double {
+      return b.max_corner().get<1>();
+    };
+
+    double globalMinX = minX(b1) < minX(b2) ? minX(b1) : minX(b2);
+    double globalMinY = minY(b1) < minY(b2) ? minY(b1) : minY(b2);
+    double globalMaxX = maxX(b1) > maxX(b2) ? maxX(b1) : maxX(b2);
+    double globalMaxY = maxY(b1) > maxY(b2) ? maxY(b1) : maxY(b2);
+
+    return {{globalMinX, globalMinY}, {globalMaxX, globalMaxY}};
+  }
+
+  static bool IsBorderOfSplitCandidate(uint64_t current, uint64_t splitSize,
+                                       uint64_t M) {
+    if (((current + 1) % splitSize == 0 && (current + 1) / splitSize < M) ||
+        (current % splitSize == 0 && current / splitSize >= 1))
+      return true;
+    return false;
+  }
+
+  // ___________________________________________________________________________
+  // Convert a single wkt literal to a datapoint in the format suitable for the
+  // Rtree
+  static std::optional<BoundingBox> ConvertWordToRtreeEntry(
+      const std::string& wkt) {
+    /**
+     * Convert a single wkt literal to a boundingbox.
+     * Get the bounding box of either a multipolygon, polygon or a linestring
+     */
+    if (!wkt.starts_with("\"MULTIPOLYGON") && !wkt.starts_with("\"POLYGON") &&
+        !wkt.starts_with("\"LINESTRING")) {
+      return {};
+    }
+
+    double maxDouble = std::numeric_limits<double>::max();
+
+    double minX = maxDouble;
+    double maxX = -maxDouble;
+    double minY = maxDouble;
+    double maxY = -maxDouble;
+
+    // Iterate over matches and capture x and y coordinates
+    for (
+        auto match : ctre::range<
+            R"( *([\-|\+]?[0-9]+(?:[.][0-9]+)?) +([\-|\+]?[0-9]+(?:[.][0-9]+)?))">(
+            wkt)) {
+      double x = std::stod(std::string(match.get<1>()));
+      double y = std::stod(std::string(match.get<2>()));
```

Comment:
If you use `absl::fromChars` then you can directly pass in string_views without converting to `std::string` (huge performance benefit).